packages/manager/apps/hpc-vmware-public-vcf-aas/src/components/form/DatacentreOrder.component.tsx

```diff
@@ -79,36 +76,48 @@ export const DatacentreOrder = <T extends OrderType>({
     min: minQuantity,
     max: maxQuantity,
   });
-  const pricedResources = getPricedVdcResources({
-    catalog: catalog?.data,
-    resources:
-      orderType === 'compute' ? orderableResource?.data?.compute : orderableResource?.data?.storage,
-  });
+
+  const pricedResources: PricedResource<T>[] = useMemo(() => {
+    if (!catalog?.data || !orderableResource?.data) return [] as PricedResource<T>[];
+    return getPricedVdcResources({
+      catalog: catalog.data,
+      resources:
+        orderType === 'compute' ? orderableResource.data.compute : orderableResource.data.storage,
+    }) as PricedResource<T>[];
+  }, [catalog, orderableResource, orderType]);
 
   useEffect(() => {
-    if (pricedResources?.length && !selectedResource) {
+    if (pricedResources[0] && !selectedResource) {
```

Comment:
The type of `pricedResources` is forced in the useMemo, accessing an index of an array on which the type is forced is a good recipe for crashes